### PR TITLE
Add `sortBy` function to `SearchBuilder` to allow for random sorting

### PR DESCRIPTION
**Pull request by dreverri**
Also update tests to take advantage of mocha's support for promises instead of using the `done` parameter.

### CODE REVIEW

File: lib/search.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ SearchBuilder.prototype.offset = function (offset) {
 SearchBuilder.prototype.sort = function (field, order) {
   assert(field, 'field required')
   assert(order, 'order required')
-  if (field.indexOf('@path.') != 0) {
+  if (field.indexOf('@path.') != 0 || field !== "_random") {
```
Comment (housejester) [inline lib/search.js]: I think this needs to be an `&&`. Otherwise, fields that have a `@path.` prefix will be also prefixed with `value.`.

**Comment by dreverri** [comment]
This will change the behavior of the client if an object has a `_random` field in the value. I think we need to removing automatic prefixing entirely and bump the version number.

**Comment by housejester** [comment]
I agree that removing the path prefix is the right approach. We should also add a RELEASE_NOTES.md like we have in the java client so we can clearly communicate breaking changes.

File: lib/search.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ SearchBuilder.prototype.offset = function (offset) {
 SearchBuilder.prototype.sort = function (field, order) {
   assert(field, 'field required')
   assert(order, 'order required')
-  if (field.indexOf('@path.') != 0) {
+  if (field.indexOf('@path.') != 0 && field !== "_random") {
```
Comment (benjismith) [inline lib/search.js]: There's one other possibility here, `_score`, which we should also specifically account for.

**Comment by dreverri** [comment]
@benjismith @housejester I've decided to introduce a new `sortBy` function and leave the current `sort` function behavior. The `sortBy` function just appends a given sort clause to a list of sort clauses that are joined together when the request is made. I also added a `sortRandom` function that simply calls `sortBy("_random")` or optionally adds a seed value (e.g. `sortRandom("seedValue")` is the same as `sortBy("_random:seedValue")`).

**Comment by dreverri** [comment]
Weird, some tests are failing on node 0.11.

File: lib/search.js [hunk 3]
```diff
@@ -61,17 +63,30 @@ SearchBuilder.prototype.offset = function (offset) {
  * @return {SearchBuilder}
  */
 SearchBuilder.prototype.sort = function (field, order) {
```
Comment (benjismith) [inline lib/search.js]: Can we add a `@deprecated` annotation to the jsdoc for the `sort` function?

**Comment by benjismith** [comment]
:+1:

**Pull request merged**